snapcraft/plugins/dotnet.py

```diff
@@ -140,3 +151,64 @@ def _get_appname(self):
             if fnmatch.fnmatch(file, '*.??proj'):
                 return os.path.splitext(file)[0]
                 break
+
+    def _get_version_metadata(self, version):
+        jsonData = self._get_dotnet_release_metadata()
+        package_data = list(filter(lambda x: x.get('version-runtime')
+                            == version, jsonData))
+
+        if not package_data or len(package_data) < 1:
+            print('error occurred while fetching  the version details'
+                  'or the version specified is incorrect')
+
+        return package_data
+
+    def _get_dotnet_release_metadata(self):
+        package_metadata = []
+
+        req = urllib.request.Request(_DOTNET_RELEASE_METADATA_URL)
+        r = urllib.request.urlopen(req).read()
+        package_metadata = json.loads(r.decode('utf-8'))
+
+        return package_metadata
+
+    def _get_sdk_info(self, version):
+        metadata = self._get_version_metadata(version)
+
+        sdk_packge_name = []
+        if 'sdk-linux-x64' in metadata[0]:
+            # look for sdk-linux-x64 property, if it doesn't exist
+            # look for ubuntu.14.04 entry as shipped during 1.1
+            sdk_packge_name = metadata[0]['sdk-linux-x64']
+        elif 'sdk-ubuntu.14.04' in metadata[0]:
+            sdk_packge_name = metadata[0]['sdk-ubuntu.14.04']
+
+        sdk_package_url = '{}{}'.format(metadata[0]['blob-sdk'],
+                                        sdk_packge_name)
+        sdk_checksum = self._get_package_checksum(
+                            metadata[0]['checksums-sdk'], sdk_packge_name)
+
+        return {'package_url': sdk_package_url, 'checksum': sdk_checksum}
+
+    def _get_package_checksum(self, checksum_url, filename):
+        req = urllib.request.Request(checksum_url)
+        r = urllib.request.urlopen(req).read()
+        data = r.decode('utf-8').split('\n')
+
+        checksum = []
+        for line in data:
+            text = line.split()
+            if len(text) == 2 and 'Hash' in text[0]:
+                hash = text[1]
+
+            if len(text) == 2 and filename in text[1]:
+                checksum = text[0]
+                break
+
+        return '{}/{}'.format(hash.lower(), checksum)
+
+    def env(self, root):
+        if root == self.installdir:
```

Comment:
This will conflict with the other .net PR, and I took issue with this part, but I won't cover it again here.

Reply:
I completely overlooked this being in here. Thanks for catching it!